pkg/virt-operator/resource/generate/install/strategy.go

```diff
@@ -244,67 +249,112 @@ func DumpInstallStrategyToConfigMap(clientset kubecli.KubevirtClient, operatorNa
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.roles {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.roleBindings {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
+	delimeter := []byte("---\n")
 	for _, entry := range strategy.crds {
-		b, _ := yaml.Marshal(entry)
-		writer.Write([]byte("---\n"))
-		writer.Write(b)
+		b, err := yaml.Marshal(entry)
+		if err != nil {
+			return nil, err
+		}
+		safeWriteToBuffer(writer, delimeter)
+		safeWriteToBuffer(writer, b)
 	}
 	for _, entry := range strategy.services {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.certificateSecrets {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.validatingWebhookConfigurations {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.mutatingWebhookConfigurations {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.apiServices {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.deployments {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.daemonSets {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.sccs {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.serviceMonitors {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.prometheusRules {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
 	}
 	for _, entry := range strategy.configMaps {
-		marshalutil.MarshallObject(entry, writer)
+		if err := marshalutil.MarshallObject(entry, writer); err != nil {
+			return nil, err
+		}
+	}
+	if err := writer.Flush(); err != nil {
+		panic(fmt.Errorf("Flush on bytes should be safe, %v", err))
```

Comment:
Again - why panic and not returning error ?

Reply:
I don't think that the error is possible but it's easy to return here.